### ISSUE TITLE
Orientation of edges in `incident_edges`

### ISSUE THREAD

**Issue by leburgel** [mtfishman/NamedGraphs.jl#11]
Currently `incident_edges(graph, vertex)` returns a list of edges which are directed away from `vertex` towards its neighbors:

https://github.com/mtfishman/NamedGraphs.jl/blob/869194950011956ca5c5b056a84550e7f898fa7c/src/Graphs/abstractgraph.jl#L41

I thought it might make more sense to reverse the direction to have the `incident_edges` point towards the `vertex` instead. Not that this is very relevant by itself, but I bumped into it when overloading this for effective Hamiltonians on trees in which case I definitely wanted to have the `incident_edges` pointing inward (https://github.com/leburgel/ITensorNetworks.jl/blob/0eb9ef98ab7d758117c3918452c9fc8831b4e914/src/treetensornetwork/abstractprojttno.jl#L27).

If there was a particular reason to have them point outward in the first place then I could just as well change it in the ITensorNetworks PR, since it would be best to stick to the same convention everywhere.

**Comment by mtfishman** [comment]
This is addressed by #19.

The design there is that I define `incident_edges(graph, vertex; dir=:out)`, where `dir ∈ (:out, :in, :both)`.

For undirected graphs, it has outputs like the following:
```julia
julia> g = path_graph(3)
{3, 2} undirected simple Int64 graph

julia> incident_edges(g, 2; dir=:out)
2-element Vector{Graphs.SimpleGraphs.SimpleEdge{Int64}}:
 Edge 2 => 1
 Edge 2 => 3

julia> incident_edges(g, 2; dir=:in)
2-element Vector{Graphs.SimpleGraphs.SimpleEdge{Int64}}:
 Edge 1 => 2
 Edge 3 => 2

julia> incident_edges(g, 2; dir=:both)
4-element Vector{Graphs.SimpleGraphs.SimpleEdge{Int64}}:
 Edge 2 => 1
 Edge 2 => 3
 Edge 1 => 2
 Edge 3 => 2
```

For directed graphs:
```julia
julia> g = path_digraph(3)
{3, 2} directed simple Int64 graph

julia> incident_edges(g, 2; dir=:out)
1-element Vector{Graphs.SimpleGraphs.SimpleEdge{Int64}}:
 Edge 2 => 3

julia> incident_edges(g, 2; dir=:in)
1-element Vector{Graphs.SimpleGraphs.SimpleEdge{Int64}}:
 Edge 1 => 2

julia> incident_edges(g, 2; dir=:both)
2-element Vector{Graphs.SimpleGraphs.SimpleEdge{Int64}}:
 Edge 2 => 3
 Edge 1 => 2
```

@leburgel open to feedback on this interface/definition.

**Comment by leburgel** [comment]
Looks perfect to me!
